src/otx/cli/utils/jsonargparse.py

```diff
@@ -351,14 +353,70 @@ def get_configuration(config_path: str | Path) -> dict:
     from otx.cli.cli import OTXCLI
 
     with patch_update_configs():
-        parser = OTXCLI.engine_subcommand_parser()
+        parser, _ = OTXCLI.engine_subcommand_parser(subcommand=subcommand)
+        if kwargs:
+            parser.set_defaults(**kwargs)
+
         args = parser.parse_args(args=["--config", str(config_path)], _skip_check=True)
+
     config = namespace_to_dict(args)
     logger.info(f"{config_path} is loaded.")
 
     # Remove unnecessary cli arguments for API usage
-    cli_args = ["verbose", "data_root", "task", "seed", "callback_monitor", "resume", "disable_infer_num_classes"]
+    cli_args = [
+        "verbose",
+        "data_root",
+        "task",
+        "seed",
+        "callback_monitor",
+        "resume",
+        "disable_infer_num_classes",
+        "workspace",
+    ]
     logger.warning(f"The corresponding keys in config are not used.: {cli_args}")
     for arg in cli_args:
         config.pop(arg, None)
     return config
+
+
+def get_instantiated_classes(
```

Comment:
@samet-akcay @ashwinvaidya17 I think this is a good example that can be applied to `AnomalibCLI` as well. We support having the API and CLI behave the same as much as possible.

Reply:
![image](https://github.com/openvinotoolkit/training_extensions/assets/38045080/9d29b6a4-967a-4a56-bbe0-876d416e4eb0)

Reply:
@harimkang, I agree

Reply:
API and CLI should ideally have the same behaviour